tests/MongoDB.Driver.Tests/Jira/CSharp3188Tests.cs

```diff
@@ -0,0 +1,94 @@
+            var serverResponseDelay = TimeSpan.FromMilliseconds(1000);
+            var appName = $"app_async_{async}";
+
+            var timeoutCommand = BsonDocument.Parse($@"
+            {{
+                configureFailPoint : 'failCommand',
+                mode : {{
+                    times : 1
+                }},
+                data : {{
+                    failCommands : ['ping'],
+                    appName : '{appName}',
+                    blockConnection : true,
+                    blockTimeMS : {serverResponseDelay.TotalMilliseconds}
+                }}
+            }}");
+            var mongoClientSettings = DriverTestConfiguration.GetClientSettings().Clone();
+            mongoClientSettings.ApplicationName = appName;
+            mongoClientSettings.SocketTimeout = socketTimeout;
+            using (var client = DriverTestConfiguration.CreateDisposableClient(mongoClientSettings))
+            {
+                using (var failPoint = FailPoint.Configure(client.Cluster, NoCoreSession.NewHandle(), timeoutCommand))
+                {
+                    var db = client.GetDatabase("db");
+                    if (async)
+                    {
+                        var exception = Record.Exception(() => db.RunCommandAsync<BsonDocument>("{ ping : 1 }").GetAwaiter().GetResult());
+
+                        var mongoConnectionException = exception.Should().BeOfType<MongoConnectionException>().Subject;
+                        mongoConnectionException.ContainsSocketTimeoutException.Should().BeFalse();
+                        mongoConnectionException.ContainsTimeoutException.Should().BeTrue();
+                        mongoConnectionException
+                            .InnerException.Should().BeOfType<TimeoutException>().Subject
+                            .InnerException.Should().BeNull();
+                    }
+                    else
+                    {
+                        var exception = Record.Exception(() => db.RunCommand<BsonDocument>("{ ping : 1 }"));
+
+                        var mongoConnectionException = exception.Should().BeOfType<MongoConnectionException>().Subject;
+                        mongoConnectionException.ContainsSocketTimeoutException.Should().BeTrue();
+                        mongoConnectionException.ContainsTimeoutException.Should().BeTrue();
+                        var socketException = mongoConnectionException
+                            .InnerException.Should().BeOfType<IOException>().Subject
+                            .InnerException.Should().BeOfType<SocketException>().Subject;
+                        socketException.SocketErrorCode.Should().Be(SocketError.TimedOut);
+                        socketException.InnerException.Should().BeNull();
+                    }
+                }
+            }
+
+            Thread.Sleep(serverResponseDelay - socketTimeout); // wait until server will reset failpoint blocking
```

Comment:
Needs to pass local tests (not sure why it's not problem for the EG). As far as I understand, the failpoint doesn't stop working until the specified server delay will be exceeded. So we're waiting this time before next test attempt

Reply:
Remove call to `Thread.Sleep`. It should not be necessary.

Reply:
done